Small fixes

Snowstar38 opened the pull request:
- Cats will be properly sorted when you open the catlist screen for the first time (fixing a rare bug with medicine cats not being in order)
- The "next" and "previous" cat buttons now work properly for the StarClan/Dark Forest guide.
- Minor spelling and pronoun tagging fixes
- Fixed tagging issue with omen list for a few patrols

Snowstar38 committed:
Fixing some sorting/display bugs

- Cats will be properly sorted when you open the catlist screen for the first time (fixing a rare bug with medicine cats not being in order)
- The "next" and "previous" cat buttons now work properly for the StarClan/Dark Forest guide.

Snowstar38 committed:
Spelling fixes

Snowstar38 committed:
Lack of break broke it

Snowstar38 committed:
Tiny pronoun tagging fix

keyraven commented on scripts/screens/base_screens.py:
```diff
@@ -169,6 +169,7 @@ def menu_button_pressed(self, event):
         elif event.ui_element == self.menu_buttons["camp_screen"]:
             self.change_screen('camp screen')
         elif event.ui_element == self.menu_buttons["catlist_screen"]:
+            Cat.sort_cats()
```
We don't want to be resorting the entire cat list whenever we open the catlist screen. This has the potential to cause lag on very large clans, and is a good amount of unneeded duplicate work. 

The bug with medicine cat may by caused be small differences in the behavior of `list.sort` and `bisect.insort`.  For now, I'm not sure how to fix this issue, and it might be more of a pain to fix then it's worth.

keyraven commented on scripts/screens/cat_screens.py:
```diff
@@ -670,26 +670,28 @@ def determine_previous_and_next_cat(self):
 
         if is_instructor:
-            next_cat = 1
+            current_cat_found = 1
```
Out of curiosity, what was causing the next-cat issue? I thought it might have been the special starclan cat list, which is what is used for the page, while the next-cat function uses the general all_cat list.

Snowstar38 committed:
Remove cat sorting

After a lot of testing, it was just.. working again with the same files. I can no longer recreate it anyways, so I guess we are fine!?

Snowstar38 committed:
Fix for s_c and p_l on 2-cat patrols

- Now, if a 2-cat patrol is tagged with rc_has_stat, p_l will not be a valid target for s_c (fixes bugs in which p_l and s_c would be the same cat)

Snowstar38 committed:
Border patrol fix

Snowstar38 committed:
Omen list patrol fix

Ryos00 commented on resources/dicts/patrols/beach/med/any.json:
```diff
@@ -156,7 +156,7 @@
         "exp": 20,
         "success_text": {
-            "unscathed_common": "p_l is plucking some rosemary when {PRONOUN/p_l/poss} attention is suddenly grabbed by a a_sign. The fur on {PRONOUN/p_l/poss} back begins to prickle slightly, the sound of {PRONOUN/p_l/poss} heartbeat like thunder in {PRONOUN/p_l/poss} ears. {PRONOUN/p_l/subject/CAP} {VERB/p_l/turn/turns} and hare back to camp, herbs forgotten. {PRONOUN/p_l/subject/CAP} {VERB/p_l ...
+            "unscathed_common": "p_l is plucking some rosemary when {PRONOUN/p_l/poss} attention is suddenly grabbed by omen_list_emotional_smell_sound. The fur on {PRONOUN/p_l/poss} back begins to prickle slightly, the sound of {PRONOUN/p_l/poss} heartbeat like thunder in {PRONOUN/p_l/poss} ears. {PRONOUN/p_l/subject/CAP} {VERB/p_l/turn/turns} and hare back to camp, herbs forgotten. {PRONOUN/p_l ...
```
I might be mistaken about how the snippet lists work, but originally the signs were something like what's in the sight section of the omen list so I think they should still be something like that.

Snowstar38 committed:
omen_list_sight

Snowstar38 committed:
A bit of pronoun tagging
